src/burn/engine/plan.cpp

```diff
@@ -1487,6 +1524,39 @@ extern "C" HRESULT PlanRelatedBundlesComplete(
             hr = DependencyPlanPackageComplete(&pRelatedBundle->package, pPlan);
             ExitOnFailure(hr, "Failed to complete plan dependency actions for related bundle package: %ls", pRelatedBundle->package.sczId);
         }
+
+        if (fInstallingAnyPackage && BOOTSTRAPPER_RELATION_UPGRADE == pRelatedBundle->relationType)
```

Comment:
1. There is no concept of "the older bundle", one bundle could have multiple bundles to uninstall as part of the Update related bundle behavior.